client/src/telemetry/glean-context.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from "react";
+import { path, referrer } from "./generated/pageView";
+import { clicked } from "./generated/element";
+import * as pings from "./generated/pings";
+import * as Glean from "@mozilla/glean/web";
+import { GLEAN_CHANNEL, GLEAN_DEBUG, GLEAN_ENABLED } from "../env";
+
+export type PageProps = {
+  referrer: string | undefined;
+  path: string | undefined;
+};
+
+export type PageEventProps = {
+  referrer: string | undefined;
+  path: string | undefined;
+};
+
+export type ElementClickedProps = {
+  source: string;
+  subscription_type: string;
+};
+
+export type GleanAnalytics = {
+  page: (arg: PageProps) => void;
+  click: (arg: ElementClickedProps) => void;
+};
+
+const FIRST_PARTY_DATA_OPT_OUT_COOKIE_NAME = "moz-1st-party-data-opt-out";
+const GLEAN_APP_ID = "mdn-yari";
+
+function glean(): GleanAnalytics {
+  if (typeof window === "undefined") {
+    //SSR return noop.
+    return {
+      page: (page: PageProps) => {},
+      click: (element: ElementClickedProps) => {},
+    };
+  }
+
+  const cookies = `; ${document.cookie};`;
+  const userIsOptedOut = cookies.includes(
+    `; ${FIRST_PARTY_DATA_OPT_OUT_COOKIE_NAME}=true;`
+  );
```

Comment:
Equivalent to this:

```suggestion
  const userIsOptedOut = document.cookie.split('; ').includes(`${FIRST_PARTY_DATA_OPT_OUT_COOKIE_NAME}=true;`);
```

Alternatively, we could just check for `document.cookie.includes(FIRST_PARTY_DATA_OPT_OUT_COOKIE_NAME)`, as opt-in removes the cookie.

Bedrock doesn't check its value either:

https://github.com/mozilla/bedrock/blob/b3b8a2bdc54b378f2800aff57e08934d3e25fb60/media/js/glean/utils.es6.js#L39